heat/naive_bayes/gaussianNB.py

```diff
@@ -0,0 +1,555 @@
+        if not isinstance(y, ht.DNDarray):
+            raise ValueError("input needs to be a ht.DNDarray, but was {}".format(type(y)))
+        if y.numdims != 1:
+            raise ValueError("expected y to be a 1-D tensor, is {}-D".format(y.numdims))
+        if sample_weight is not None:
+            if not isinstance(sample_weight, ht.DNDarray):
+                raise ValueError(
+                    "sample_weight needs to be a ht.DNDarray, but was {}".format(
+                        type(sample_weight)
+                    )
+                )
+        classes = ht.unique(y, sorted=True)
+        if classes.split is not None:
+            classes = ht.resplit(classes, axis=None)
+
+        return self.__partial_fit(X, y, classes, _refit=True, sample_weight=sample_weight)
+
+    def __check_partial_fit_first_call(self, classes=None):
+        """
+        Private helper function for factorizing common classes param logic
+        Estimators that implement the ``partial_fit`` API need to be provided with
+        the list of possible classes at the first call to partial_fit.
+        Subsequent calls to partial_fit should check that ``classes`` is still
+        consistent with a previous value of ``clf.classes_`` when provided.
+        This function returns True if it detects that this was the first call to
+        ``partial_fit`` on ``clf``. In that case the ``classes_`` attribute is also
+        set on ``clf``.
+        """
+        if getattr(self, "classes_", None) is None and classes is None:
+            raise ValueError("classes must be passed on the first call " "to partial_fit.")
+
+        elif classes is not None:
+            unique_labels = classes
+            if getattr(self, "classes_", None) is not None:
+                if not ht.equal(self.classes_, unique_labels):
+                    raise ValueError(
+                        "`classes={}` is not the same as on last call "
+                        "to partial_fit, was: {}".format(classes, self.classes_)
+                    )
+
+            else:
+                # This is the first call to partial_fit
+                self.classes_ = unique_labels
+                return True
+
+        # classes is None and self.classes_ has already previously been set:
+        # nothing to do
+        return False
+
+    @staticmethod
+    def __update_mean_variance(n_past, mu, var, X, sample_weight=None):
+        """
+        Compute online update of Gaussian mean and variance.
+        Given starting sample count, mean, and variance, a new set of
+        points X, and optionally sample weights, return the updated mean and
+        variance. (NB - each dimension (column) in X is treated as independent
+        -- you get variance, not covariance).
+        Can take scalar mean and variance, or vector mean and variance to
+        simultaneously update a number of independent Gaussians.
+        See Stanford CS tech report STAN-CS-79-773 by Chan, Golub, and LeVeque:
```

Comment:
Move to a references section. Use IEEE standard for the citation as well please. Also if this is the same paper as referenced in the main docs for GNB then you can only cite it there and point to it from here.

Reply:
Added references both in main gnb documentation and in __update_mean_variance as I suspect the latter will be needed in other classifiers as well and move out to independent module eventually.